### PR TITLE
Travis conf update and new ucsc_tools package

### PR DESCRIPTION
**Pull request by bgruening**
update to latest bioconda-utils and infrastructure changes

### CODE REVIEW

**Comment by bgruening** [comment]
@jmchilton @nsoranzo included is a large ucsc_tools package.
Do we want to ship all tools or just the ones we know? 
We can easily strip them out from here on.

**Comment by jmchilton** [comment]
I'm fine with this - seems like a good plan. Thanks a bunch for putting this together - I still don't think I have ever written a Conda package.

**Comment by bgruening** [comment]
The downside is, it can be a really large dependency.

**Comment by bgruening** [comment]
ping @yhoogstrate

**Comment by yhoogstrate** [comment]
If the purpose of this conda channel to assist the Galaxy backend, I would limit it to only those required for Galaxy (Docker images would become unnecessary large). If this becomes a more general purpose conda channel it would fit better to include all tools, but then maybe it would even better fit bioconda?

File: sorted-r-blacklist [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# TODO: cap-mirseq has test error. Perhaps just needs a fix on expected exit
```
Comment (nsoranzo) [inline sorted-r-blacklist]: Is this file needed?

**Comment by nsoranzo** [comment]
I'm with @yhoogstrate on this.

File: build-fail-blacklist [hunk 2]
```diff
@@ -0,0 +1,2 @@
+recipes/mysqlclient
+recipes/ensembl-vep
```
Comment (nsoranzo) [inline build-fail-blacklist]: This file can be removed.

File: config.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+env_matrix: "scripts/env_matrix.yml"
+blacklists:
+    - "sorted-r-blacklist"
+    - "build-fail-blacklist"
```
Comment (nsoranzo) [inline config.yml]: We're not using these ATM.

File: scripts/env_matrix.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+CONDA_PY:
+  - 27
+  - 34
+  - 35
+CONDA_BOOST: "1.60"
```
Comment (nsoranzo) [inline scripts/env_matrix.yml]:
```
CONDA_HTSLIB: "1.4"
CONDA_BOOST: "1.61"
```

File: recipes/ucsc_tools/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    - ucsc-twobitinfo ==332
+    - ucsc-twobitmask ==332
+    - ucsc-twobittofa ==332
+    - ucsc-validatefiles ==332
+    - ucsc-validatemanifest ==332
+    - ucsc-wigcorrelate ==332
+    - ucsc-wigtobigwig ==332
+    - ucsc-wordline ==332
+    - ucsc-xmlcat ==332
+    - ucsc-xmltosql ==332
```
Comment (nsoranzo) [inline recipes/ucsc_tools/meta.yaml]: The only needed ones should be:
```
    - ucsc-bedgraphtobigwig ==332
    - ucsc-bedtobigbed ==332
    - ucsc-fatotwobit ==332
    - ucsc-nibfrag ==332
    - ucsc-twobittofa ==332
    - ucsc-wigtobigwig ==332
```

We also need `ucsc-liftover`, whose last Bioconda version is 324 and was removed in https://github.com/bioconda/bioconda-recipes/commit/7b6e7801704f10f1cc614f4b737efad9f3c8beec (I think it should be re-added).

**Comment by nsoranzo** [comment]
@bgruening I've updated to the latest bioconda-recipes infrastructure and reduced `ucsc_tools` meta-package to the packages used in the tools/converters distributed with Galaxy. Travis is green, but I'm not sure if/where the built packages/containers will be pushed when this will be merged to `master`, can you check please?

**Comment by bgruening** [comment]
Oh so sorry, I completely forgot about this one :(
We just use an anaconda-token, which I have stored in the travis credentials. This token specifies also the repository afaik. So we should be ready, fingers crossed.

**Comment by nsoranzo** [comment]
quay.io ?

**Comment by bgruening** [comment]
You mean the auth? Hopefully this will work as well, yes.

**Comment by nsoranzo** [comment]
Cool, thanks! Should we press the magic button then?

**Comment by bgruening** [comment]
Please do so :)

**Comment by nsoranzo** [comment]
It worked! https://anaconda.org/iuc/ucsc_tools
